### PR TITLE
AbstractDeath: Fixing Cause Generation Typo

### PR DESCRIPTION
**Pull request by Windchild292**
This fixes the cause generation and doesn't load causes when random death isn't enabled (no need to waste the memory)

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/personnel/death/AbstractDeath.java [hunk 1]
```diff
@@ -64,7 +64,9 @@ protected AbstractDeath(final RandomDeathMethod method, final CampaignOptions op
         setUseRandomPrisonerDeath(options.isUseRandomPrisonerDeath());
         this.enableRandomDeathSuicideCause = options.isUseRandomDeathSuicideCause();
         this.causes = new HashMap<>();
-        initializeCauses();
+        if (!method.isNone()) {
```
Comment (rjhancock) [inline MekHQ/src/mekhq/campaign/personnel/death/AbstractDeath.java:67]: Possible to add an opposite version of this for easier reading?

File: MekHQ/src/mekhq/campaign/personnel/death/AbstractDeath.java [hunk 2]
```diff
@@ -252,11 +254,12 @@ private void initializeCausesFromFile(final File file) {
             if (!wn.hasChildNodes()) {
                 continue;
             }
+
             try {
                 final Gender gender = Gender.valueOf(wn.getNodeName());
                 getCauses().putIfAbsent(gender, new HashMap<>());
                 final NodeList nl2 = wn.getChildNodes();
-                for (int j = 0; j < nl.getLength(); j++) {
+                for (int j = 0; j < nl2.getLength(); j++) {
```
Comment (rjhancock) [inline MekHQ/src/mekhq/campaign/personnel/death/AbstractDeath.java:262]: Can we get better variable names other than `nl2` and `nl`? I know they mean `NodeList` bust still...

**Review by rjhancock — APPROVED** [review]

File: MekHQ/src/mekhq/campaign/personnel/death/AbstractDeath.java (hunk 1 above)
Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/personnel/death/AbstractDeath.java:67]: I personally find it to be harder to read, which is why I've coded it this way.

File: MekHQ/src/mekhq/campaign/personnel/death/AbstractDeath.java (hunk 2 above)
Comment (Windchild292) [reply MekHQ/src/mekhq/campaign/personnel/death/AbstractDeath.java:262]: It's a legacy naming setup, and one I don't want to fix currently. It is annoying, but standardization here is better for now.

Comment (rjhancock) [reply MekHQ/src/mekhq/campaign/personnel/death/AbstractDeath.java:262]: Ugh. Maybe one day.

**Pull request merged**